imageio/imageio-metadata/src/main/java/com/twelvemonkeys/imageio/metadata/tiff/TIFFReader.java

```diff
@@ -135,7 +135,9 @@ protected Directory readDirectory(final ImageInputStream pInput, final long pOff
                 CompoundDirectory next = (CompoundDirectory) readDirectory(pInput, nextOffset, true);
 
                 for (int i = 0; i < next.directoryCount(); i++) {
-                    ifds.add((IFD) next.getDirectory(i));
+                    if(next.getDirectory(i).size() > 0) {
```

Comment:
I did not consider there could be reasons for having empty directories.
For returning an empty `CompoundDirectory` I had to change a unit test, please look into that. 
`readSubdirectories` is creating an IFD array in this case, which caused a ClassCastException